frontend/views/pages/GroupSettings.vue

```diff
@@ -181,6 +181,14 @@ export default ({
       } else {
         this.openProposal('GroupLeaveModal')
       }
+    },
+    refreshForm () {
+      const { groupName, sharedValues, mincomeCurrency } = this.groupSettings
+      this.form = {
+        groupName,
+        sharedValues,
+        mincomeCurrency
+      }
```

Comment:
I also don't understand which issue this is fixing and what it's doing.

Could you add a comment explaining in detail which issue this is fixing and why it works?

Reply:
In the previous way, the `form` variable is initiated once only when the component is created.
So when user switch group, the form is not updated to the details of new group so it displays the wrong data.
What I did here is to update this form every time the `groupSettings` is changed so that it can display the current group details.

Reply:
Got it! Could you add a comment to that effect then?

Something along the lines of:

```js
// re-fetch the latest correct values whenever the user switches groups
```